allennlp/modules/token_embedders/bag_of_word_counts_token_embedder.py

```diff
@@ -0,0 +1,80 @@
+import torch
+from allennlp.modules.token_embedders.token_embedder import TokenEmbedder
+from allennlp.data import Vocabulary
+from allennlp.common import Params
+from allennlp.nn.util import get_text_field_mask
+
+
+@TokenEmbedder.register("bag_of_word_counts")
+class BagOfWordCountsTokenEmbedder(TokenEmbedder):
+    """
+    Represents a sequence of tokens as a bag of (discrete) word ids, as it was done
+    in the pre-neural days.
+    
+    Each sequence gets a vector of length vocabulary size, where the i'th entry in the vector
+    corresponds to number of times the i'th token in the vocabulary appears in the sequence.
+
+    Parameters
+    ----------
+    vocab_size: int
+        size of vocabulary
+    projection_dim : int, optional (default = None)
+        if specified, will project the resulting bag of words representation
+        to specified dimension.
+    """
+    def __init__(self, vocab_size: int, projection_dim: int = None) -> None:
+        super(BagOfWordCountsTokenEmbedder, self).__init__()
+        self.vocab_size = vocab_size
+        if projection_dim:
+            self._projection = torch.nn.Linear(vocab_size, projection_dim)
+        else:
+            self._projection = None
+        self.output_dim = projection_dim or vocab_size
+
+    def get_output_dim(self):
+        return self.num_embeddings
```

Comment:
This isn't correct if you have a projection - should be `return self.output_dim`.